packages/react-ui-core/src/Counter/__tests__/Counter-test.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import Counter from '../Counter'
+
+const theme = {}
+const leftOperatorClick = function (component) {
+  const incrementValue = component.state.count + 1
+  component.setState({
```

Comment:
Calling `setState` should only be done in the component, itself.

If you want to decouple the actual count from the `Counter` component then you should pass in a `count` prop and have `Counter` render using that, and keep the `count` state outside of the component. However, I can't think of a reason you would really need to specialize your increment/decrement to do anything other than go down/up by one. Maybe there is a use case there, but this may be a case of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

What I would probably do is do the `setState` in the `increment()` or `decrement()` of `Counter` component. This way, the user of `Counter` doesn't have to implement state handling and the component is useful right away. On top of that, you can easily add an `onIncrement` or `onDecrement` prop callback to override the default behavior later on.